File: app/frontend/entities/Asset.ts
interface Asset {
  id: number;
  symbol: string;
  quantity: number;
  currency?: string;
  cost: number;
  market_value?: number;
  marketCost?: number;
  marketValue?: number;
  sector: string;
  category: string;
  last_price_update?: string;
  current_price?: number;
}

export default Asset;

File: app/models/asset.rb
class Asset < ApplicationRecord
  # Asset represents a unique holding in the portfolio
  # Aggregated view of all operations for a given symbol
  validates :symbol, presence: true, uniqueness: true
  validates :quantity, :cost, :market_value, numericality: { greater_than_or_equal_to: 0 }
end
